machine/dell/dell_s4000_c2338/i2ctools/i2ctools-sys-eeprom-platform.patch

```diff
@@ -0,0 +1,255 @@
+ 	return(FALSE);
+-    case TLV_CODE_VENDOR_EXT:
+     default:
+ 	if (set_bytes(data, strval, &new_tlv_len) != 0 ) {
+ 	    return(FALSE);
+@@ -723,6 +717,7 @@ bool tlvinfo_add_tlv(u_int8_t * eeprom, int tcode, char * strval)
+     return(TRUE);
+ }
+ 
++#define MAX_EEPROM_I2C_BUS_PATH 15
+ /*
+  * read_sys_eeprom - read the hwinfo from i2c EEPROM
+  */
+@@ -733,9 +728,16 @@ int read_sys_eeprom(void *eeprom_data, int offset, int len)
+     int i = 0;
+     u_int8_t *c;
+     int addr = SYS_EEPROM_OFFSET + offset;
++    int i2c_addr, i2c_bus;
++    char i2c_bus_path[MAX_EEPROM_I2C_BUS_PATH];
+ 
+     c = eeprom_data;
+-    if (eeprom_open(SYS_EEPROM_I2C_DEVICE, SYS_EEPROM_I2C_ADDR,
++    i2c_addr = get_i2c_addr();
++    i2c_bus = get_i2c_bus();
++    memset(i2c_bus_path, 0, MAX_EEPROM_I2C_BUS_PATH);
++    snprintf(i2c_bus_path, MAX_EEPROM_I2C_BUS_PATH, "/dev/i2c-%d", i2c_bus);
++
++    if (eeprom_open(i2c_bus_path, i2c_addr,
+ 		    EEPROM_TYPE_8BIT_ADDR, &e)) {
+ 	printf("ERROR: Cannot open I2C device\n");
+ 	return -1;
+@@ -758,15 +760,23 @@ int write_sys_eeprom(void *eeprom_data, int len)
+     int i = 0;
+     u_int8_t *c;
+     u_int16_t  addr = SYS_EEPROM_OFFSET;
++    int i2c_addr, i2c_bus;
++    char i2c_bus_path[MAX_EEPROM_I2C_BUS_PATH];
+ 
+     c = eeprom_data;
++    i2c_addr = get_i2c_addr();
++    i2c_bus = get_i2c_bus();
++    memset(i2c_bus_path, 0, MAX_EEPROM_I2C_BUS_PATH);
++    snprintf(i2c_bus_path, MAX_EEPROM_I2C_BUS_PATH, "/dev/i2c-%d", i2c_bus);
++
+     for (i = 0; i < len; i++) {
+-	if (eeprom_open(SYS_EEPROM_I2C_DEVICE, SYS_EEPROM_I2C_ADDR,
++	if (eeprom_open(i2c_bus_path, i2c_addr,
+ 			EEPROM_TYPE_8BIT_ADDR, &e)) {
+ 	    printf("ERROR: Cannot open I2C device\n");
+ 	    return -1;
+ 	}
+ 	eeprom_write_byte(&e, addr, *c);
++	usleep(5000);
+ 	eeprom_close(&e);
+ 	c++; addr++;
+     }
+diff --git a/sys_eeprom/onie_tlvinfo.h b/sys_eeprom/onie_tlvinfo.h
+index 5714f37..da8ab3a 100644
+--- a/sys_eeprom/onie_tlvinfo.h
++++ b/sys_eeprom/onie_tlvinfo.h
+@@ -170,3 +170,5 @@ bool tlvinfo_delete_tlv(u_int8_t * eeprom, u_int8_t code);
+ bool tlvinfo_add_tlv(u_int8_t * eeprom, int tcode, char * strval);
+ bool tlvinfo_decode_tlv(u_int8_t *eeprom, u_int8_t tcode, char* value);
+ void show_tlv_code_list(void);
++int get_i2c_addr(void);
++int get_i2c_bus (void);
+diff --git a/sys_eeprom/sys_eeprom.c b/sys_eeprom/sys_eeprom.c
+index a77837f..2c57a58 100644
+--- a/sys_eeprom/sys_eeprom.c
++++ b/sys_eeprom/sys_eeprom.c
+@@ -5,6 +5,9 @@
+ #include "onie_tlvinfo.h"
+ 
+ static u_int8_t eeprom[SYS_EEPROM_SIZE];
++static int i2c_addr = SYS_EEPROM_I2C_ADDR;
++static int i2c_bus = 0;
++
+ /*
+  *  This macro defines the sys_eeprom command line command.
+  */
+@@ -12,10 +15,14 @@ cmd_usage()
+ {
+     static const char *usage =
+ 	"Display and program the system EEPROM data block.\n"
+-	"Usage: sys-eeprom [-h][-l] [-e] [-s <code>=<value>,...]\n"
++	"Usage: sys-eeprom [-h][-l] [-e] [-b bus] [-a address] [-s <code>=<value>,...]\n"
```

Comment:
What is this for?

Does this mean you have multiple ONIE TLV compliant EEPROMs on the board?

That in and of itself does not really present a problem.